src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,139 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75f;
+    private static final int RESIZE_COEFFICIENT = 2;
+    private Node<K, V>[] myHashMap;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        myHashMap = new Node[INITIAL_CAPACITY];
+        threshold = (int) (myHashMap.length * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        putNode(newNode);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        V value = null;
+        for (Node<K, V> node : myHashMap) {
+            if (node == null) {
+                continue;
+            }
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    value = node.value;
+                }
+                node = node.next;
+            }
+        }
+        return value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    public boolean containsKey(K key) {
+        for (Node<K, V> node : myHashMap) {
+            if (node == null) {
+                continue;
+            }
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    return true;
+                }
+                node = node.next;
+            }
+        }
+        return false;
+    }
+
+    public boolean containsValue(V value) {
+        for (Node<K, V> node : myHashMap) {
+            if (node == null) {
+                continue;
+            }
+            while (node != null) {
+                if (Objects.equals(value, node.value)) {
+                    return true;
+                }
+                node = node.next;
+            }
+        }
+        return false;
+    }
+
+    private void resize() {
+        Node<K, V>[] oldHashMap = myHashMap;
+        myHashMap = new Node[oldHashMap.length * RESIZE_COEFFICIENT];
+        threshold = (int) (myHashMap.length * LOAD_FACTOR);
+        for (Node<K, V> node : oldHashMap) {
+            if (node == null) {
+                continue;
+            }
+            while (node != null) {
+                putNode(node);
+                node = unlink(node);
+            }
+        }
+    }
+
+    private Node<K, V> unlink(Node<K, V> node) {
+        Node<K, V> currentNode = node.next;
+        node.next = null;
+        return currentNode;
+    }
+
+    private void putNode(Node<K, V> node) {
+        int bucket = Math.abs(node.hash % myHashMap.length);
```

Comment:
maybe a separate method? for example, 
`private int getIndex(K key) {
return key == null ? 0 : Math.abs(node.hash % myHashMap.length);
}`

Reply:
No, Tamara uses this code only in this method